chat-android/src/main/java/com/ably/chat/Occupancy.kt

```diff
@@ -60,20 +71,127 @@ data class OccupancyEvent(
 )
 
 internal class DefaultOccupancy(
-    private val messages: Messages,
+    realtimeChannels: AblyRealtime.Channels,
+    private val chatApi: ChatApi,
+    private val roomId: String,
+    private val logger: Logger,
 ) : Occupancy {
-    override val channel: Channel
-        get() = messages.channel
+    private val messagesChannelName = "$roomId::\$chat::\$chatMessages"
+
+    override val channel: Channel = realtimeChannels.get(messagesChannelName, ChatChannelOptions {
+        params = mapOf(
+            "occupancy" to "metrics"
+        )
+    })
+
+    private val listeners: MutableList<Occupancy.Listener> = CopyOnWriteArrayList()
+
+    private val eventBus = MutableSharedFlow<OccupancyEvent>(
```

Comment:
This has a good throttling use-case, for both typing and occupancy events

Reply:
Typing also seems to be a valid use-case https://medium.com/@appdevinsights/use-of-the-debounce-operator-in-kotlin-flows-0b05ae892a82